### ISSUE TITLE
[Truffle] Sprintf issue in Rack

### ISSUE THREAD

**Issue by bjfish** [jruby/jruby#2981]
Found in Rack logger. Used by Sinatra.

Expected
```
ruby -e 'puts %{%s - %s [%s] "%s %s%s %s" %d %s %0.4f\n} % ["::1", "-", "24/May/2015:15:09:29 +0000", "GET", "/hi", "", "HTTP/1.1", "200", "14", 0.07400000000000001]'
::1 - - [24/May/2015:15:09:29 +0000] "GET /hi HTTP/1.1" 200 14 0.0740
```
Actual
```
$ jt e 'puts %{%s - %s [%s] "%s %s%s %s" %d %s %0.4f\n} % ["::1", "-", "24/May/2015:15:09:29 +0000", "GET", "/hi", "", "HTTP/1.1", "200", "14", 0.07400000000000001]'
core:/core/string.rb:12:in `sprintf': Truffle doesn't have a case for the org.jruby.truffle.pack.nodes.type.ToIntegerNodeGen node with values of type  String(org.jruby.truffle.runtime.core.RubyString) (TypeError)
	from core:/core/string.rb:12:in `%'
	from -e:1:in `<main>'
```

**Comment by chrisseaton** [comment]
Note that Webrick may return bad time values at the moment, as I shimmed `httpdate`. Unrelated to this bug but thought I'd mention.

**Comment by chrisseaton** [comment]
Minimal test case:

```ruby
puts sprintf("%d", "200")
```

**Comment by chrisseaton** [comment]
Our printf isn't well tested at the moment, due to https://github.com/ruby/rubyspec/issues/68.

**Comment by chrisseaton** [comment]
Fixed in 3b0bd45351d13f0f89c83d36ac110b3f6b8a462d and e5384d1ad4128a93443c6986ec42add4a22f62b8
